docs/core/magnetic_field.md

```diff
@@ -0,0 +1,228 @@
+# Magnetic field
+
+The magnetic field component of ACTS provides functionality to describe
+arbitrary magnetic field configurations in an experiment. It is implemented in
+a generic way and can be extended to connect to an experiment specific upstream
+source of field data.
+
+Algorithms which need magnetic field information (e.g.
+{class}`Acts::AtlasStepper`, {class}`Acts::EigenStepper`) accept the magnetic
+field as an explicit argument. 
+
+## Provider interface
+
+All magnetic field implementations inherit and implement the interface {class}`Acts::MagneticFieldProvider`:
+
+
+It provides a generic interface over different implementations. To speed up
+magnetic field lookup, each implementation can have a *cache* object, which the
+client is expected to pass into lookup calls:
+
+:::{doxygentypedef} Acts::MagneticFieldProvider::Cache
+:::
+
+The implementation is then free to use and update this cache instance as
+needed. Before a client can issue field lookup calls, it needs to obtain an
+initialized instance of this cache object. This can be achieved generically for
+all implementations by using {func}`Acts::MagneticFieldProvider::makeCache`.
+This function accepts an instance of {type}`Acts::MagneticFieldContext`, see
+[](#magnetic-field-context) for details.
+
+The main lookup method of {class}`Acts::MagneticFieldProvider` is
+
+:::{doxygenfunction} Acts::MagneticFieldProvider::getField
+:outline:
+:::
+
+Aside from the lookup position
+as a global vector, it accepts an instance of the opaque cache object mentioned
```

Comment:
global position vector (I mean the vector can be local variable - right?)

Reply:
Yeah, I'm referring to the coordinate system.

Reply:
Changed.